llama_index/vector_stores/postgres.py

```diff
@@ -416,9 +420,9 @@ def _query_with_score(
                     node_id=item.node_id,
                     text=item.text,
                     metadata=item.metadata_,
-                    similarity=(1 - distance) if distance is not None else 0,
+                    similarity=(1 - item.distance) if item.distance is not None else 0,
```

Comment:
ok you calc the similarity here instead of during querying time.

Reply:
Correct, the query result is actually inverted, smaller score means more similar. IIRC @logan decided to invert them programatically a long time ago

And yes, I also tried to invert the score on query time, but that resulted in a far slower query, because postgres need to calculate  each row's score on every query.

Reply:
Found it, here's the discussion back then.

https://github.com/run-llama/llama_index/issues/7214